modules/3d/include/opencv2/3d.hpp

```diff
@@ -468,77 +468,308 @@ can be found in:
+
+
+    /** @brief Structure to keep LevMarq settings
+
+    The structure allows a user to pass algorithm parameters along with their names like this:
+    @code
+    MySolver solver(nVars, callback, MySolver::Settings().geodesicS(true).geoScale(1.0));
+    @endcode
+
+    */
+    struct CV_EXPORTS Settings
+    {
+        Settings() :
+            jacobiScaling(false),
+            upDouble(true),
+            useStepQuality(true),
+            clampDiagonal(true),
+            stepNormInf(false),
+            checkRelEnergyChange(true),
+            checkMinGradient(true),
+            checkStepNorm(true),
+            geodesic(false),
+            hGeo(1e-4),
+            geoScale(0.5),
+            stepNormTolerance(1e-6),
+            relEnergyDeltaTolerance(1e-6),
+            minGradientTolerance(1e-6),
+            smallEnergyTolerance(0), // not used by default
+            maxIterations(500),
+            initialLambda(0.0001),
+            initialLmUpFactor(2.0),
+            initialLmDownFactor(3.0)
+        { }
+
+        bool operator==(const Settings& other) const
+        {
+            const double eps = FLT_EPSILON;
+            bool ok = true;
+            ok = ok && (this->jacobiScaling == other.jacobiScaling);
+            ok = ok && (this->upDouble == other.upDouble);
+            ok = ok && (this->useStepQuality == other.useStepQuality);
+            ok = ok && (this->clampDiagonal == other.clampDiagonal);
+            ok = ok && (this->stepNormInf == other.stepNormInf);
+            ok = ok && (this->checkRelEnergyChange == other.checkRelEnergyChange);
+            ok = ok && (this->checkMinGradient == other.checkMinGradient);
+            ok = ok && (this->checkStepNorm == other.checkStepNorm);
+
+            ok = ok && (this->geodesic == other.geodesic);
+            ok = ok && abs(this->hGeo - other.hGeo) < eps;
+            ok = ok && abs(this->geoScale - other.geoScale) < eps;
+
+            ok = ok && abs(this->stepNormTolerance - other.stepNormTolerance) < eps;
+            ok = ok && abs(this->relEnergyDeltaTolerance - other.relEnergyDeltaTolerance) < eps;
+            ok = ok && abs(this->minGradientTolerance - other.minGradientTolerance) < eps;
+            ok = ok && abs(this->smallEnergyTolerance - other.smallEnergyTolerance) < eps;
+
+            ok = ok && (this->maxIterations == other.maxIterations);
+
+            ok = ok && abs(this->initialLambda - other.initialLambda) < eps;
+            ok = ok && abs(this->initialLmUpFactor - other.initialLmUpFactor) < eps;
+            ok = ok && abs(this->initialLmDownFactor - other.initialLmDownFactor) < eps;
+
+            return ok;
+        }
+
+        Settings& jacobiScalingS          (bool         v) { jacobiScaling           = v; return *this; }
+        Settings& upDoubleS               (bool         v) { upDouble                = v; return *this; }
+        Settings& useStepQualityS         (bool         v) { useStepQuality          = v; return *this; }
```

Comment:
Why is here 'S' suffix?
Where are you find that?

Reply:
S stands for "set" but I didn't want to give them names like `setEpsilon` to make method names more compact.
Since you recommended not to use underscores, I seek other ways to name them.

Reply:
Decided to use `setValue` names instead, looks more natural